pkg/monitortests/network/onpremhaproxy/monitortest.go

```diff
@@ -0,0 +1,291 @@
+
+	return localRecorder.Intervals(time.Time{}, time.Time{}), nil, nil
+}
+
+func (*operatorLogAnalyzer) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	constructedIntervals := monitorapi.Intervals{}
+
+	allHaproxyChanges := startingIntervals.Filter(func(eventInterval monitorapi.Interval) bool {
+		if eventInterval.Message.Reason == monitorapi.OnPremHaproxyStatusChange {
+			return true
+		}
+		return false
+	})
+
+	// for every node we draw how it sees all the other nodes, so NxN array, 3x3 for 3-node cluster
+	haproxyChanges := map[string][]monitorapi.Interval{}
+	for _, change := range allHaproxyChanges {
+		myKey := fmt.Sprintf("%s___%s", change.Locator.Keys[monitorapi.LocatorNodeKey], change.Message.Annotations[monitorapi.AnnotationNode])
+		haproxyChanges[myKey] = append(haproxyChanges[myKey], change)
+	}
+
+	for key, allChangesForTheNode := range haproxyChanges {
+		previousChangeTime := time.Time{}
+		createdIntervals := monitorapi.Intervals(allChangesForTheNode).Filter(func(eventInterval monitorapi.Interval) bool {
+			return eventInterval.Message.Reason == monitorapi.OnPremHaproxyStatusChange
+		})
+		if len(createdIntervals) > 0 {
+			previousChangeTime = createdIntervals[0].From
+		}
+		lastAvailableStatus := ""
+
+		for _, availableConditionChange := range allChangesForTheNode {
+			currentStatus := availableConditionChange.Message.Annotations[monitorapi.AnnotationStatus]
+			if currentStatus == lastAvailableStatus {
+				continue
+			}
+
+			if currentStatus == "UP" && lastAvailableStatus == "DOWN" {
+				constructedIntervals = append(constructedIntervals,
+					monitorapi.NewInterval(monitorapi.SourceHaproxyMonitor, monitorapi.Info).
+						//Locator(availableConditionChange.Locator).
+						Locator(monitorapi.Locator{Keys: map[monitorapi.LocatorKey]string{
+							monitorapi.LocatorOnPremKubeapiUnreachableFromHaproxyKey: key,
+						}}).
+						Message(monitorapi.NewMessage().Reason(monitorapi.OnPremHaproxyDetectsDown).
+							Constructed(monitorapi.ConstructionOwnerOnPremHaproxy).
+							HumanMessage(fmt.Sprintf("Kubeapi on %s is detected dead by %s", availableConditionChange.Message.Annotations[monitorapi.AnnotationNode], availableConditionChange.Locator.Keys["node"]))).
+						Display().
+						Build(previousChangeTime, availableConditionChange.From),
+				)
+			}
+
+			previousChangeTime = availableConditionChange.From
+			lastAvailableStatus = availableConditionChange.Message.Annotations[monitorapi.AnnotationStatus]
+		}
+
+		//deletionTime := time.Now()
+		//deletedIntervals := monitorapi.Intervals(allChangesForTheNode).Filter(func(eventInterval monitorapi.Interval) bool {
+		//	return eventInterval.Message.Reason == monitorapi.APIServiceDeletedInAPI
+		//})
+		//if len(deletedIntervals) > 0 {
+		//	deletionTime = deletedIntervals[0].To
+		//}
+		//if len(lastAvailableStatus) > 0 {
```

Comment:
Something like this check is needed so that if the job run ends with the `lastAvailableStatus == "DOWN"`, there is an interval on the timeline.